### PR TITLE
Updated dio version to ^5.0.0

### PR DESCRIPTION
**Pull request by OnlyTarg**
closes: [conflict with dio 5.0.1 #47](https://github.com/alippo-com/GrowthBook-SDK-Flutter/issues/47)

### CODE REVIEW

**Comment by DK070202** [comment]
Hi @OnlyTarg ,

Thank you so much for your contribution to our open-source project! We're thrilled to see new contributors joining us and helping improve the project.

Your effort to update the dependency and add a note to the changelog is greatly appreciated. Your contribution will definitely enhance the project's functionality and stability.

Before we proceed with merging your changes, we kindly ask you to consider the following steps:

1. **Update SDK Version**: Please update the SDK version.

2. **Changelog Update**: It would be great if you could provide a brief description of the changes you've made in the project's `changelog.md` file. This helps other users and contributors understand the evolution of the project over time.

Feel free to reach out if you have any questions or need further assistance. Once you've made the requested changes, please comment on the pull request so that we can re-review it.

**Comment by OnlyTarg** [comment]
Hi @DK070202 
Done.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,6 @@
+# 2.1.1+0
+- Fixes [issue](https://github.com/alippo-com/GrowthBook-SDK-Flutter/issues/47)
+
 # 2.1.0+0
```
Comment (DK070202) [inline CHANGELOG.md:4]: Add one more hash in front:

```diff
- # 2.1.0+0
+ ## 2.1.0+0 
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,6 @@
+# 2.1.1+0
+- Fixes [issue](https://github.com/alippo-com/GrowthBook-SDK-Flutter/issues/47)
```
Comment (DK070202) [inline CHANGELOG.md:2]: As it updates the Dio version to the major we should consider updating the SDK version to the major. 
```md 
# 3.0.0+0
```

File: pubspec.yaml [hunk 3]
```diff
@@ -1,13 +1,13 @@
 name: growthbook_sdk_flutter
 description: An open-source feature flagging and experimentation platform that makes it simple to alter features and execute A/B testing.
-version: 2.1.0+0
+version: 2.1.1+0
```
Comment (DK070202) [inline pubspec.yaml:3]: As it updates the Dio version to the major we should consider updating the SDK version to the major.

**Review by DK070202 — CHANGES_REQUESTED** [review]

**Comment by DK070202** [comment]
> Hi @DK070202 Done.

Thanks for the quick updates.
Now it's just two few changes away from the master.

File: CHANGELOG.md (hunk 1 above)
Comment (OnlyTarg) [reply CHANGELOG.md]: Done.

File: CHANGELOG.md (hunk 2 above)
Comment (OnlyTarg) [reply CHANGELOG.md]: Done.

File: pubspec.yaml (hunk 3 above)
Comment (OnlyTarg) [reply pubspec.yaml]: Done.

**Comment by DK070202** [comment]
LGTM!

**Review by DK070202 — APPROVED** [review]

**Pull request merged**